ci/visual_tests.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""Execute visual regression tests against a running django server."""
+
+from percy import percy_snapshot
+from selenium import webdriver
+
+# For docs on Percy's python/selenium integration, see:
+# https://docs.percy.io/docs/python-selenium
+
+
+def setup() -> webdriver.Chrome:
+    """Initialize and return a browser driver to use for taking snapshots."""
+    options = webdriver.ChromeOptions()
+    options.add_argument("--no-sandbox")
+    options.add_argument("--disable-extensions")
+    options.add_argument("--disable-dev-shm-usage")
+    options.add_argument("--disable-setuid-sandbox")
+    options.add_argument("--headless")
+    return webdriver.Chrome(options=options)
+
+
+def run():
+    """Take DOM snapshots of a set of URLs and upload to Percy."""
+    browser = setup()
+
+    # homepage
+    browser.get("http://localhost:8000/")
```

Comment:
Would it make sense to have a list of urls that should be tested? (Or I suppose a dict with url and name). It seems like that would be easier and more readable to manage, although I'm not sure how you'd handle the urls that require the toggle click interaction — maybe that's could be a separate list, since they are all handled the same way (and would probably all change when/if we revise the nav filter logic).

Reply:
i had a list originally (although dict makes sense to cover the titles too) but got rid of it for that reason (the ones that require extra interaction). i guess the question is if we expect to add any other kinds of interaction on any pages — for e.g. PPA I can see testing _lots_ of interactions, especially on the search page, so it can get complicated (and specific to an individual page) quickly.

one thing i could do here is just always try to "click" the dropdown even if the page doesn't have one — i don't think selenium will error, it just won't do anything for that particular page. so that way there only needs to be one "visit" function that covers all the pages

Reply:
Or just two loops, one that requires an interaction and one that doesn't?

Reply:
I think I'm going to leave this as it is for now, since we will soon write tests for a page (or pages) that will likely have new types of interaction not covered here — the search page. I don't think we know enough yet about how best to refactor this, and it may continue to get complicated as we test more behaviors.